examples_tests/23.Autoexposure/main.cpp

```diff
@@ -4,20 +4,771 @@
+	{
+		return windowCb->isWindowOpen();
+	}
 
+private:
+	core::smart_refctd_ptr<nbl::ui::IWindow> window;
+	core::smart_refctd_ptr<CommonAPI::CommonAPIEventCallback> windowCb;
+	core::smart_refctd_ptr<nbl::video::IAPIConnection> apiConnection;
+	core::smart_refctd_ptr<nbl::video::ISurface> surface;
+	core::smart_refctd_ptr<nbl::video::IUtilities> utilities;
+	core::smart_refctd_ptr<nbl::video::ILogicalDevice> logicalDevice;
+	video::IPhysicalDevice* physicalDevice;
+	std::array<video::IGPUQueue*, CommonAPI::InitOutput::MaxQueuesCount> queues;
+	video::IGPUQueue* graphicsComputeQueue;
+	core::smart_refctd_ptr<nbl::video::ISwapchain> swapchain;
+	core::smart_refctd_ptr<video::IGPURenderpass> renderpass = nullptr;
+	std::array<nbl::core::smart_refctd_ptr<nbl::video::IGPUFramebuffer>, CommonAPI::InitOutput::MaxSwapChainImageCount> fbos;
+	core::smart_refctd_ptr<nbl::system::ISystem> system;
+	core::smart_refctd_ptr<nbl::asset::IAssetManager> assetManager;
+	video::IGPUObjectFromAssetConverter::SParams cpu2gpuParams;
+	core::smart_refctd_ptr<nbl::system::ILogger> logger;
+	core::smart_refctd_ptr<CommonAPI::InputSystem> inputSystem;
+	video::IGPUObjectFromAssetConverter cpu2gpu;
+
+	int32_t resourceIx = -1;
+	uint32_t acquiredNextFBO = {};
+
+	uint32_t outBufferIx[SC_IMG_COUNT] = { 0u, 0u, 0u };
+	std::chrono::high_resolution_clock::time_point lastPresentStamp;
+
+	core::smart_refctd_ptr<video::IGPUSemaphore> imageAcquire[FRAMES_IN_FLIGHT] = { nullptr };
+	core::smart_refctd_ptr<video::IGPUSemaphore> renderFinished[FRAMES_IN_FLIGHT] = { nullptr };
+	core::smart_refctd_ptr<video::IGPUFence> frameComplete[FRAMES_IN_FLIGHT] = { nullptr };
+	core::smart_refctd_ptr<video::IGPUCommandBuffer> cmdbufs[FRAMES_IN_FLIGHT] = { nullptr };
+
+	core::smart_refctd_ptr<video::IGPUImageView> imageToTonemapView = nullptr;
+	core::smart_refctd_ptr<video::IGPUImageView> quantStorageImageViews[SC_IMG_COUNT] = { nullptr };
+	core::smart_refctd_ptr<video::IGPUImageView> tonemapperOutImageViews[SC_IMG_COUNT] = {};
+	core::smart_refctd_ptr<video::IGPUBuffer> lumaMeterUbo = nullptr;
+	core::smart_refctd_ptr<video::IGPUBuffer> toneParamsLumaOutputBuffer[SC_IMG_COUNT] = {};
+
+	ToneMapperClass::Params_t<TMO> tonemappingParams = ToneMapperClass::Params_t<TMO>(Exposure, Key, 0.85f);
+
+	core::smart_refctd_ptr<video::IGPUDescriptorSet> lumaMeteringDS[SC_IMG_COUNT] = { };
+	core::smart_refctd_ptr<video::IGPUDescriptorSet> tonemappingDS[SC_IMG_COUNT] = { };
+	core::smart_refctd_ptr<video::IGPUComputePipeline> lumaMeteringPipeline = nullptr;
+	core::smart_refctd_ptr<video::IGPUComputePipeline> tonemappingPipeline = nullptr;
+	LumaMeterClass::DispatchInfo_t lumaDispatchInfo;
+
+	core::smart_refctd_ptr<video::IGPUDescriptorSet> fstDS[SC_IMG_COUNT] = {};
+	core::smart_refctd_ptr<nbl::video::IGPUGraphicsPipeline> fstGraphicsPipeline = nullptr;
+};
+
+NBL_COMMON_API_MAIN(AutoexposureSampleApp)
+
+extern "C" {  _declspec(dllexport) DWORD NvOptimusEnablement = 0x00000001; }
+
+#if 0
```

Comment:
delete old code

Reply:
@nipunG314 we no longer need the stuff in the `if0` and the `extern "C"` (that was for OpenGL to choose NV always, in VK you can just choose GPUs)